tests/test_merge.py

```diff
@@ -145,3 +149,77 @@ def test_get_average_frame_count(samples: str, count: float) -> None:
+                process_text="python-123/123;[Profiling error: exception CalledProcessError] 1",
+            ),
+            ";python;[Profiling error: exception CalledProcessError];[unknown];_PyEval_EvalFrameDefault;"
+            "(/usr/local/lib/libpython3.6m.so.1.0);(/usr/local/lib/libpython3.6m.so.1.0) 3",
+            id="1perf_1pyspy-error",
+        ),
+        pytest.param(
+            dict(
+                perf_text="python-123/123;_PyObject_Call_Prepend;_PyObject_FastCallDict;_PyFunction_FastCallDict 2\n"
+                "python-123/123;_start;__libc_start_main;main;Py_Main;PyRun_SimpleFileExFlags 3\n"
+                "python-123/123;entry_SYSCALL_64_[k] 1",
+                process_text="python-123/123;[Profiling error: exception CalledProcessError] 1",
+            ),
+            ";python;[Profiling error: exception CalledProcessError];_PyObject_Call_Prepend;_PyObject_FastCallDict;"
+            "_PyFunction_FastCallDict 2\n"
+            ";python;[Profiling error: exception CalledProcessError];_start;__libc_start_main;main;Py_Main;"
+            "PyRun_SimpleFileExFlags 3\n"
+            ";python;[Profiling error: exception CalledProcessError];entry_SYSCALL_64_[k] 1",
+            id="3perf_1pyspy-error",
+        ),
+        pytest.param(
+            dict(
+                perf_text="java-123/123;(/opt/java/lib/libjvm.so);__vdso_gettimeofday;(vdso);apic_timer_interrupt_[k];"
+                "smp_apic_timer_interrupt_[k] 4",
+                process_text="java-123/123;[Profiling skipped: async-profiler is already loaded] 1",
+            ),
+            ";java;[Profiling skipped: async-profiler is already loaded];(/opt/java/lib/libjvm.so);__vdso_gettimeofday;"
+            "(vdso);apic_timer_interrupt_[k];smp_apic_timer_interrupt_[k] 4",
+            id="1perf_1java-skipped",
+        ),
+        pytest.param(
+            dict(
+                perf_text="python-123/123;EMPTY 0",
+                process_text="python-123/123;[Profiling error: exception CalledProcessError] 1",
+            ),
+            "",
+            id="empty-perf_1pyspy-error",
+        ),
+    ],
+)
+def test_merge_profiles_onto_errors(input_dict: Dict[str, str], expected: str) -> None:
+    enrichment_options = EnrichmentOptions(
+        profile_api_version=None,
+        container_names=False,
+        application_identifiers=False,
+        application_identifier_args_filters=[],
+        application_metadata=False,
+    )
+    metadata: Metadata = dict(profiling_mode="cpu")
+    metrics = Metrics(cpu_avg=1.0, mem_avg=1.0)
+    perf_pid_to_profiles = parse_profiles_text(input_dict["perf_text"])
+    process_profiles = parse_profiles_text(input_dict["process_text"])
+    outcome_raw = merge_profiles(perf_pid_to_profiles, process_profiles, None, enrichment_options, metadata, metrics)
+    header_outcome = outcome_raw.split("\n", maxsplit=1)
+    header, outcome = header_outcome[0], header_outcome[1] if len(header_outcome) >= 2 else ""
```

Comment:
```suggestion
    header, outcome = header_outcome[0], header_outcome[1] if len(header_outcome) == 2 else ""
```
you expect 1 or 2 so keep code explicit

Reply:
Sure.